### PR TITLE
Added introduction to HTTP to programming skills section

### PR DESCRIPTION
**Pull request by tbtommyb**
Introduction to HTTP, focusing on methods and status codes. Added TODO section for further development.

### CODE REVIEW

File: programmer-skills/http-intro.md [hunk 1]
```diff
@@ -0,0 +1,361 @@
+Introduction to HTTP
+====================
+
+The Hypertext Transfer Protocol (HTTP) is the foundation of web communication, so some familiarity with it is essential for any web developer. HTTP works on the basis of a *client* sending a *request* to a *server*, which in return provides a *response*. You can think of the client as being the user's web browser and the server as being any kind of back-end server you might write (e.g. in Express.js).
```
Comment (alwinsol) [inline programmer-skills/http-intro.md]: Reads better "Hypertext Transfer Protocol (HTTP) is one of the main foundations of web communication.  As a developer it is prudent to be familiar with it.  HTTP works on the basis of a _client_ sending a _request_ to a _server_, which in return provides a _response_.  For example, a web browser is a client program at the user computer that may access information at any web server in the world. "

File: programmer-skills/http-intro.md [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+Secure authentication
+
+Cookies
+
+Sessions
+
+REST
+
+Sockets
+
```
Comment (alwinsol) [inline programmer-skills/http-intro.md:360]: Add a references section please

File: programmer-skills/http-intro.md [hunk 3]
```diff
@@ -0,0 +1,361 @@
+Introduction to HTTP
+====================
+
+The Hypertext Transfer Protocol (HTTP) is the foundation of web communication, so some familiarity with it is essential for any web developer. HTTP works on the basis of a *client* sending a *request* to a *server*, which in return provides a *response*. You can think of the client as being the user's web browser and the server as being any kind of back-end server you might write (e.g. in Express.js).
+
+Browsers must formulate valid requests with the appropriate information for that request. Sevrvers will respond to requests according to the specifications. Browsers will in turn interpret these responses according to the specifications. this means that it is important that both your front-end and back-end code is using HTTP correctly.
```
Comment (alwinsol) [inline programmer-skills/http-intro.md]: Reads better "Servers respond to requests according to specifications. Browsers interpret these specific responses. So it imperative that front and back-end HTTP is configured correctly"

File: programmer-skills/http-intro.md [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+
+Of course, as so often happens in web development, different browsers have their own implementation quirks.
+
```
Comment (alwinsol) [inline programmer-skills/http-intro.md:9]: Change to "Competing browsers behave differently with their own individual behaviour."

File: programmer-skills/http-intro.md [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+Further reading
+===============
+
+https://httpstatuses.com/
+
+http://code.tutsplus.com/tutorials/http-headers-for-dummies--net-8039
+
+[The actual specification](https://www.w3.org/Protocols/rfc2616/rfc2616.txt), if you're feeling brave!
+
+TODO
```
Comment (alwinsol) [inline programmer-skills/http-intro.md:343]: Is the TODO section to be left in or taken out?

Comment (tbtommyb) [inline programmer-skills/http-intro.md:343]: I have seen other sections that have a TODO section at the bottom and I think it's a useful indicator for anyone looking for something to contribute, so I was planning to keep it in.

**Pull request closed without merging**